src/shogun/evaluation/SplittingStrategy.h

```diff
@@ -99,14 +96,14 @@ class CSplittingStrategy: public CSGObject
 	CLabels* m_labels;
 
 	/** subset indices */
-	CDynamicObjectArray* m_subset_indices;
+	mutable CDynamicObjectArray* m_subset_indices;
```

Comment:
I see now what your thinking was.
I don't think this is the best style, making it mutable and the other things const.
There are simpler ways of reaching this goal imo